### PR TITLE
Feature/add shipments index endpoint

### PR DESCRIPTION
**Pull request by Alonso2105**
# Description

What did you implement/Why did you implement this?

- The shipments_controller controller was created to apply the shipments filtering logic.
- Made a migration to add a default value in the role field in the user model.
- A before action was added to authenticate the user if he is not.

## Evidence
you can run rails s in console inside the shipments_tracker project and navigate to: http://127.0.0.1:3000/shipments/index
Screenshot 2023-08-11 at 17 26 34
Screenshot 2023-08-11 at 17 47 54

## Screenshots

### CODE REVIEW

File: shipments_tracker/app/controllers/shipments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      carrier = Carrier.find_by(name: permitted_params[:carrier_name])
+      @shipments = @shipments.where(carrier: carrier)
+    end
+  end
+
+  private
+
+  def permitted_params
+    params.permit!(:carrier_name, :status)
+  end
```
Comment (JuanequeX) [inline shipments_tracker/app/controllers/shipments_controller.rb]: where is the end of the class?

Indented correctly at 2 spaces, please

File: shipments_tracker/app/helpers/shipments_helper.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module ShipmentsHelper
+  def tracking_status(status)
+    I18n.t("shipments.status.#{status.downcase}")
+  end
+end
+
+# @shipments.each do |shipment|
+#   tracking_status(shipment.status)
+# end
```
Comment (JuanequeX) [inline shipments_tracker/app/helpers/shipments_helper.rb]: delete this code comment

File: shipments_tracker/app/models/shipment.rb [hunk 3]
```diff
@@ -5,11 +5,9 @@ class Shipment < ApplicationRecord
   # Validación: El campo status no puede estar vacío y solo puede contener los valore especificados
   validates :status, presence: true, inclusion: { in: %w(REGISTERED IN_TRANSIT DELAYED LOST STOLEN DELIVERED) }
 
-  # Validación: El campo tracking_history no puede estar vacío y debe tener al menos un elemento (longitud mínima de 1)
-  validates :tracking_history, presence: true, length: { minimum: 1 }
```
Comment (JuanequeX) [inline shipments_tracker/app/models/shipment.rb]: this same change is apply for victor in this [PR](https://github.com/BrightCoders-Institute/proyecto-final-ror-team1/pull/25/files#diff-65021091b0c955d78481490438cb3351dbba5144846c4b5b4c145b9b1e11d582L9)

File: shipments_tracker/test/controllers/shipments_controller_test.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  #   assert_redirected_to shipment_url(@shipment)
+  # end
+
+  # test "should destroy shipment" do
+  #   assert_difference("Shipment.count", -1) do
+  #     delete shipment_url(@shipment)
+  #   end
+
+  #   assert_redirected_to shipments_url
+  # end
```
Comment (JuanequeX) [inline shipments_tracker/test/controllers/shipments_controller_test.rb]: please, remove any commented code that is unnecessary within the assessment to solve in the PR

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: shipments_tracker/app/controllers/shipments_controller.rb (hunk 1 above)
Comment (Alonso2105) [reply shipments_tracker/app/controllers/shipments_controller.rb]: corrected

File: shipments_tracker/test/controllers/shipments_controller_test.rb (hunk 4 above)
Comment (Alonso2105) [reply shipments_tracker/test/controllers/shipments_controller_test.rb]: corrected

File: shipments_tracker/app/helpers/shipments_helper.rb (hunk 2 above)
Comment (Alonso2105) [reply shipments_tracker/app/helpers/shipments_helper.rb]: corrected

File: shipments_tracker/app/models/shipment.rb (hunk 3 above)
Comment (Alonso2105) [reply shipments_tracker/app/models/shipment.rb:9]: ni entiendo bien, ¿que debo corregir ?

Comment (Alonso2105) [reply shipments_tracker/app/models/shipment.rb]: Done!

**Review by JuanequeX — APPROVED** [review]

**Pull request merged**